lib/cyclonedx/cocoapods/bom_builder.rb

```diff
@@ -119,10 +120,10 @@ def add_to_bom(xml)
     class BOMBuilder
       NAMESPACE = 'http://cyclonedx.org/schema/bom/1.4'
 
-      attr_reader :component, :pods
+      attr_reader :component, :pods, :dependencies
 
-      def initialize(component: nil, pods:)
-        @component, @pods = component, pods
+      def initialize(component: nil, dependencies: nil, pods:)
```

Comment:
Codacy is asking you to put `pods:` as the first parameter because `pods` is the only parameter that does not specify a default value.  Calls to this function can then look like one of the following (note that `pods` is always first because it has to be specified):

```
BOMBuilder.new(pods: pods)
BOMBuilder.new(pods: pods, component: myComponent)
BOMBuilder.new(pods: pods, dependencies: deps)
BOMBuilder.new(pods: pods, component: myComponent, dependencies: deps)
```